addons/website_event_sale/controllers/main.py

```diff
@@ -70,3 +71,15 @@ def registration_confirm(self, event, **post):
                 request.website.sale_reset()
 
         return res
+
+
+class PaymentPortalOnsite(PaymentPortal):
+
+    def _validate_transaction_for_order(self, transaction, sale_order_id):
+        """
+        Throws a ValidationError if the user tries to pay for a ticket which isn't available
+        """
+        super()._validate_transaction_for_order(transaction, sale_order_id)
+        sale_order = request.env['sale.order'].browse(sale_order_id).exists().sudo()
+        tickets = sale_order.order_line.filtered(lambda line: line.product_type == 'event').mapped('event_ticket_id')
+        tickets._check_seats_availability(minimal_availability=sale_order.attendee_count)
```

Comment:
:thinking: it would be better to count per ticket I think. Otherwise if you buy 2x2 tickets, you won't be able to do it if any of the tickets unless all the tickets have 4 spots left, instead of 2 each. (would be nice to add that case to the test :D)